src/Moq/Invocation.cs

```diff
@@ -161,5 +195,13 @@ public override string ToString()
 
 			return builder.ToString();
 		}
+
+		private enum VerificationState : byte
+		{
+			Invoked = 0,
+			InvokedAndMatchedBySetup,
+			InvokedAndMatchedByVerifiableSetup,
+			Verified,
```

Comment:
Note that the logic implemented in the new `Invocation` methods ensures that we only ever progress from `Invoked` towards `Verified`, but it should never be possible to regress back to an earlier flag.